src/api/index.tsx

```diff
@@ -105,6 +107,8 @@ export enum HttpMethod {
 type CallApiParams = {
   _makeQueryString?: typeof makeQueryString;
   apiState: ApiState;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  bodyData?: Record<string, any>;
```

Comment:
It could be literally anything: number, string, array, Object. There is probably a better way of providing type safety by typing the endpoint payload and passing that type through a generic. I'll see if I can do that.

Reply:
Actually, yeah, this will be a lot safer with a generic type (and it was easy to implement).